src/Nethermind/Nethermind.State/PersistentStorageProvider.cs

```diff
@@ -36,7 +36,7 @@ internal sealed class PersistentStorageProvider : PartialStorageProviderBase
         private readonly ResettableDictionary<StorageCell, byte[]> _originalValues = new();
 
         private readonly ResettableHashSet<StorageCell> _committedThisRound = new();
-        private readonly Dictionary<StorageCell, byte[]> _blockCache = new(4_096);
+        private readonly Dictionary<AddressAsKey, Dictionary<StorageCell, byte[]>> _blockCache = new(4_096);
```

Comment:
As what is saved to tree is only the index `tree.Set(change.StorageCell.Index, change.Value)`

Looks like it can be:

```suggestion
        private readonly Dictionary<AddressAsKey, Dictionary<UInt256, byte[]>> _blockCache = new(4_096);
```

Rather than using the much larger `Address|UInt256|bool` tuple that StorageKey is